src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,196 @@
+package duke.storage;//package duke;
+
+import duke.task.DeadlineTask;
+import duke.task.EventTask;
+import duke.task.Task;
+import duke.task.TodoTask;
+
+import java.io.*;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * The Storage class manages the loading and saving of tasks to a file.
+ */
+public class Storage {
+    String filePath;
+    ArrayList<Task> tasks;
+
+    /**
+     * Constructs a Storage object with the given file path.
+     *
+     * @param filePath the file path where tasks are stored
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Validates if the start date is before the end date.
+     *
+     * @param startDate the start date
+     * @param endDate   the end date
```

Comment:
Should consider capitalising the first letter of the description sentence. Should include a punctuation behind the parameter description too, according to the [`Comments` section](https://se-education.org/guides/conventions/java/intermediate.html#comments) of the Java Coding Standard.